src/fns/unconstrained_ops.nr

```diff
@@ -171,34 +171,40 @@ pub(crate) unconstrained fn __udiv_mod<let N: u32>(
     let mut divisor_u60: U60Repr<N, 2> = U60Repr::from(divisor);
     let b = divisor_u60;
 
-    let mut bit_difference = remainder_u60.get_msb() - divisor_u60.get_msb();
-
-    let mut accumulator_u60: U60Repr<N, 2> = U60Repr::one();
-    divisor_u60 = divisor_u60.shl(bit_difference);
-    accumulator_u60 = accumulator_u60.shl(bit_difference);
-
-    if (divisor_u60.gte(remainder_u60 + U60Repr::one())) {
-        divisor_u60.shr1();
-        accumulator_u60.shr1();
-    }
-    for _ in 0..(N * 120) {
-        if (remainder_u60.gte(b) == false) {
-            break;
+    let numerator_msb = remainder_u60.get_msb();
+    let divisor_msb = divisor_u60.get_msb();
+    if divisor_msb > numerator_msb {
+        ([0; N], numerator)
+    } else {
+        let mut bit_difference = numerator_msb - divisor_msb;
+
+        let mut accumulator_u60: U60Repr<N, 2> = U60Repr::one();
+        divisor_u60 = divisor_u60.shl(bit_difference);
+        accumulator_u60 = accumulator_u60.shl(bit_difference);
+
+        if (divisor_u60.gte(remainder_u60 + U60Repr::one())) {
```

Comment:
```suggestion
        // divisor > remainder
        if (!remainder_u60.gte(divisor_u60)) {
```

lhs >= rhs +1 => lhs > rhs => !(lhs <= rhs) => !(rhs >= lhs)

This is more efficient.